### ISSUE TITLE
Enum initialized by value is not detected as an Enum

### ISSUE THREAD

**Issue by ironfoot** [microsoft/pyright#1068]
**Describe the bug**
When an instance variable is an enum type, pyright doesn't think `variable(value)` is an enum.

**To Reproduce**
```python
import enum
from typing import Any, Type


class Color(enum.Enum):
    RED = 1
    GREEN = 2

c: enum.Enum = Color(1)  # This is fine.


class Foo:
    enum_type: Type[enum.Enum]

    def __init__(self):
        self.enum_type = Color

    def red(self) -> enum.Enum:
        return self.enum_type(1)  # Detected as error, shown below.
```

```
Expected enum item string as second parameterPylance
Expression of type "Type[enum]" cannot be assigned to return type "Enum"
  "Type[EnumMeta]" is incompatible with "Type[Enum]"Pylance (reportGeneralTypeIssues)
```

**Expected behavior**
No error in both cases.

**VS Code extension or command-line**
Pylance VS Code extension version: 2020.9.6

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.76, which I just published. It will also be included in the next release of Pylance.
